src/Repository/DefaultUnleashRepository.php

```diff
@@ -109,6 +109,7 @@ public function getFeatures(): iterable
                     $response = $this->httpClient->sendRequest($request);
                     if ($response->getStatusCode() === 200) {
                         $data = (string) $response->getBody();
+                        json_decode($data, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
Sadly it needs to work on 7.2 as well, which doesn't support the `JSON_THROW_ON_ERROR` flag. This needs to be checked manually using [json_last_error](https://www.php.net/manual/en/function.json-last-error.php).

I also don't like the idea that decoding json would be done twice, so it would be best to change the `parseFeatures()` method to accept already `json_decode`d array instead of a string.